envoy/upstream/upstream.h

```diff
@@ -1013,6 +1013,9 @@ class ClusterInfo : public Http::FilterChainFactory {
    */
   virtual const envoy::config::cluster::v3::Cluster::CommonLbConfig& lbConfig() const PURE;
 
+  virtual absl::optional<uint64_t>
+  processHttpForOutlierDetection(Http::ResponseHeaderMap& reponse) const PURE;
```

Comment:
```suggestion
  virtual absl::optional<bool>
  checkFailureResponse(Http::ResponseHeaderMap& reponse) const PURE;
```

return an `absl::optional<bool>` make more sense. It's unecessary to return http code.

Reply:
Agree.